server/models/Member.ts

```diff
@@ -24,55 +25,114 @@ const invalidateContributorsCacheUsingInstance = instance => {
   return null;
 };
 
-interface MemberModelStaticInterface {
-  isActive(member: { tier?: Tier; lastDonation?: Date }): boolean;
-  connectCollectives(
-    childCollective: Collective,
-    parentCollective: Collective,
-    user: User,
-    memberInfo: any,
-    options?: { transaction?: SequelizeTransaction },
-  ): Promise<MemberModelInterface>;
-}
-
-export interface MemberModelInterface
-  extends Model<InferAttributes<MemberModelInterface>, InferCreationAttributes<MemberModelInterface>> {
-  id: number;
-  CreatedByUserId: number;
-  MemberCollectiveId: number;
-  CollectiveId: number;
-  TierId: number;
-  role: roles;
-  description: string;
-  publicMessage: string;
+class Member extends Model<InferAttributes<Member, { omit: 'info' }>, InferCreationAttributes<Member>> {
+  declare public readonly id: CreationOptional<number>;
+  declare public CreatedByUserId: ForeignKey<User['id']>;
+  declare public MemberCollectiveId: ForeignKey<Collective['id']>;
+  declare public CollectiveId: ForeignKey<Collective['id']>;
+  declare public TierId: ForeignKey<Tier['id']>;
+  declare public role: roles;
+  declare public description: string;
+  declare public publicMessage: string;
+  declare public since: Date;
+  declare public createdAt: Date;
+  declare public updatedAt: Date;
+  declare public deletedAt?: Date;
+
+  declare public tier?: Tier;
+  declare public memberCollective?: Collective;
+  declare public collective?: Collective;
```

Comment:
The way to type these is usually:

```suggestion
  declare public tier?: BelongsToGetAssociationMixin<Tier>;
  declare public memberCollective?: BelongsToGetAssociationMixin<Collective>;
  declare public collective?: BelongsToGetAssociationMixin<Collective>;
```

Is this missing bit the reason why you had to do:
```diff
- member.collective = collective
+ member['collective'] = collective
```

in https://github.com/opencollective/opencollective-api/pull/10793/files#diff-e0233b2efe7b6d82ca4f3d789b6c2cc07e74b88d94a3eec9bdcd51f058dcdfe9?

Reply:
Thanks for the comment. These are actually declared with `NonAttribute<>`, `BelongsToGetAssociationMixin<>` is used for `getTier` and `getCollective` declarations.

Reply:
True, my bad!